gf2_192/src/gf2_192poly.rs

```diff
@@ -0,0 +1,242 @@
+//!  By Leonid Reyzin
+//!  This is free and unencumbered software released into the public domain.
+//!
+//!  Anyone is free to copy, modify, publish, use, compile, sell, or
+//!  distribute this software, either in source code form or as a compiled
+//!  binary, for any purpose, commercial or non-commercial, and by any
+//!  means.
+//!
+//!  In jurisdictions that recognize copyright laws, the author or authors
+//!  of this software dedicate any and all copyright interest in the
+//!  software to the public domain. We make this dedication for the benefit
+//!  of the public at large and to the detriment of our heirs and
+//!  successors. We intend this dedication to be an overt act of
+//!  relinquishment in perpetuity of all present and future rights to this
+//!  software under copyright law.
+//!
+//!  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+//!  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+//!  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//!  IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+//!  OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+//!  ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+//!  OTHER DEALINGS IN THE SOFTWARE.
+//!
+//!  For more information, please refer to <http://unlicense.org>
+
+use thiserror::Error;
+
+use crate::{gf2_192::Gf2_192, Gf2_192Error};
+
+/// Byte representation of the coefficients of `Gf2_192Poly`.
+pub struct CoefficientsByteRepr<'a> {
+    /// Coefficient of constant term of degree zero.
+    pub coeff0: [i8; 24],
+    /// Coefficients of the non-zero-degree terms, starting with degree 1.
+    pub more_coeffs: &'a [i8],
+}
+
+#[derive(PartialEq, Eq, Clone, Debug)]
+/// A polynomial with coefficients in GF(2^192), whose domain ranges from `[i8::MIN, i8::MAX]`.
+pub struct Gf2_192Poly {
+    /// Coefficients of the polynomial. Must be non-empty.
+    coefficients: Vec<Gf2_192>,
+    /// Upper bound on the degree of the polynomial.
+    degree: usize,
+}
+
+/// `Gf2_192Poly` error
+#[derive(Error, PartialEq, Eq, Debug, Clone)]
+pub enum Gf2_192PolyError {
+    /// `Gf2_192Poly` interpolation error
+    #[error("`Gf2_192Poly::interpolation`: `points.len() != values.len()`")]
+    InterpolatePointsAndValuesLengthDiffer,
+}
+
+impl Gf2_192Poly {
+    /// Create the unique lowest-degree interpolating polynomial that passes through
+    /// `(0, value_at_zero)` and `(points[i], values[i])` for all `i = 0, ..(points.len() - 1)`.
+    /// Assumptions:
+    ///  - Elements of `points` must be distinct and must not contain `0`.
+    ///  - `points.len() == values.len()`. Note that `points` and `values` can be empty, resulting
+    ///    in a constant polynomial with value `value_at_zero`.
+    pub fn interpolate(
```

Comment:
I implemented a simplified form of this method from `sigmastate-interpreter` where `value_at_zero` must be present. This is in line with your indicated usage in `ergotree-interpreter`.

Reply:
I've added some more explanations; it covers everything from `sigmastate-interpreter`. Please let me know if it needs more.